src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -890,112 +903,107 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
                                     StringCchCopy(sourceName, ARRAYSIZE(sourceName), originalName);
                                 }
 
-                                wchar_t newReplaceTerm[MAX_PATH] = { 0 };
                                 PWSTR replaceTerm = nullptr;
-                                SYSTEMTIME LocalTime;
+                                SYSTEMTIME fileTime = {0};
 
-                                if (SUCCEEDED(spRenameRegEx->GetReplaceTerm(&replaceTerm)) && isFileAttributesUsed(replaceTerm))
+                                if (!useFileTime || ( SUCCEEDED(spItem->GetTime(&fileTime)) && SUCCEEDED(spRenameRegEx->PutFileTime(fileTime))))
```

Comment:
This "if" condition is not very clear. 
It would better to check `if (useFileTime)` and have a cleaner logic afterward.